tokio/src/lib.rs

```diff
@@ -430,14 +430,23 @@
 //! Enabling any other feature (including `full`) will cause a compilation
 //! failure.
 //!
+//! Note that not all API's are guaranteed to work correctly.
+//! We document some specific issues below.
+//! All failures should be easy to detect, such as panics or runtime errors, silent failure is considered a bug.
+//!
 //! The `time` module will only work on `WASM` platforms that have support for
 //! timers (e.g. wasm32-wasi). The timing functions will panic if used on a `WASM`
 //! platform that does not support timers.
 //!
-//! Note also that if the runtime becomes indefinitely idle, it will panic
-//! immediately instead of blocking forever. On platforms that don't support
+//! Note also that if the runtime becomes indefinitely idle, that will lead to errors
+//! instead of blocking forever. On platforms that don't support
 //! time, this means that the runtime can never be idle in any way.
```

Comment:
As far as I'm concerned, this paragraph explains why your use of `LocalSet` worked, and your use of the runtime directly failed. In one case, you caused a situation where the application would block forever. With `LocalSet`, your code did not trigger such a situation.

Reply:
So if I understand this, LocalSet doesn't block ever, only the runtimes may?

Reply:
It sounds like you executed the `LocalSet` using a different runtime from Tokio. In that case, the difference is due to use of that other runtime.

Reply:
Yes. I'm just not neccessarily aware of what parts of Tokio do what precisely.

And yeah, in that case I guess I use LocalSet similar to what FuturesUnordered would do. However I can use it without managing a reference to that object myself, and potentially with other niceties